`DialogPrimitive` with `Modal` and `Flyout` (non-breaking) - Documentation (#2213)
hashicorp/design-system: website/docs/components/modal/partials/version-history/4.6.0.md

## 4.6.0

### Deprecated

> In this version we have replaced the existing Modal sub-components with the equivalent sub-components from the [DialogPrimitive](/utilities/dialog-primitive) component (shared with the [Modal](/components/modal) component). They're not used anymore and for this reasons we consider them deprecated, and they'll be removed in our next major release.

Reviewer: ```suggestion
In this version, the Modal sub-components have been deprecated and replaced with the equivalent [DialogPrimitive](/utilities/dialog-primitive) sub-components (shared also with the [Flyout](/components/flyout) component). The deprecated Modal sub-components will be removed in the next major release.
```

Author: Much better! 😀

To do the migration, if you use the `<Hds::Modal::***>` invocation:

| Deprecated:           | Replace with:                  |
|-----------------------|--------------------------------|
| `Modal::Header`      | `DialogPrimitive::Header`      |
| `Modal::Body`        | `DialogPrimitive::Body`        |
| `Modal::Footer`      | `DialogPrimitive::Footer`      |

If instead you use the `(component hds/modal/***)`:

| Deprecated:              | Replace with:                     |
|--------------------------|-----------------------------------|
| `hds/modal/header`      | `hds/dialogprimitive/header`      |
| `hds/modal/body`        | `hds/dialogprimitive/body`        |
| `hds/modal/footer`      | `hds/dialogprimitive/footer`      |

> In the refactoring we have been able to maintain the existing CSS class names, to avoid breaking existing tests and CSS overrides in the consumers' codebases.

Reviewer: ```suggestion
In the refactoring, we have been able to maintain existing CSS class names to avoid breaking current tests and CSS overrides in consumers' codebases.
```
